eucaconsole/tweens.py

```diff
@@ -78,6 +82,25 @@ def tween(request):
     return tween
 
 
+def invalidate_session_cookie_tween_factory(handler, registry):
+    def tween(request):
+        try:
+            response = handler(request)
+            return response
+        except BeakerException as bex:
+            if bex.message == 'Invalid signature':
+                session_cookie = registry.settings.get('session.key', 'eucaconsole_session')
+                if session_cookie in request.cookies and \
+                        'invalidate-session' not in request.params:
```

Comment:
I'm having trouble understanding why the invalidate-session param is needed. Are there cases where a client would ignore the Set-Cookie header from the response (to delete that cookie key)?

Reply:
If there was a mismatch with the path or domain for the cookie it would not be cleared. I would not expect to hit that path, but err on the side of caution for changes in a maintenance release.